src/ViewListingDisplay/components/Listing.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+
+import PostCommentsDisplay from '../../PostCommentsDisplay/components/PostCommentsDisplay.jsx';
+
+import '../styles/Listing.css';
+
+/** Functional Component that creates a listing. Takes in listingInfo as a prop 
+(which holds the title, description, listingPrice, etc.) to fill in the listing */
+function Listing(props) {
+    const { 
+        title, 
+        description, 
+        listingPrice, 
+        numRooms, 
+        numSingles, 
+        numShared, 
+        singlePrice, 
+        sharedPrice, 
+        leaseType, 
+        startDate, 
+        endDate, 
+        numBathrooms 
+    } = props.listingInfo;
+
+    const roomInfo = getRoominfo(numSingles, numShared, singlePrice, sharedPrice);
+    const leaseTypeText = (leaseType === "YEAR_LONG") ? "Yearly" : "Monthly";
+    return (
+        <div className="listing-box">
+            <h1 className="title">{title}</h1>
+            <p className="description">{description}</p>
+            <p className="listingPrice">Listing Price: {listingPrice}</p>
+            <p className="leaseType">Lease Type: {leaseTypeText} Lease</p>
+            <div className="avaiability-container">
+                <h3>Availability</h3>
+                <p>Start: {startDate}</p>
+                <p>End: {endDate}</p>
+            </div>
+            <div className="rooms-info-containter">
+                <h3>Room(s) Information:</h3>
+                <p>Total number of room(s): {numRooms}</p>
+                { roomInfo }
+                <p>Total number of bathroom(s): {numBathrooms}</p>
+            </div>
+            <PostCommentsDisplay/>
+        </div>
+    );
+}
+
+function getRoominfo(numSingles, numShared, singlePrice, sharedPrice){
+    if((numSingles === '0' || numShared === 'USD 0') && (singlePrice === '0' || sharedPrice === 'USD 0')){
+        return "No rooms available";
+    }
```

Comment:
I think if numSingles = 0 and singlePrice = 0 this would say there's no rooms available regardless of the number of shared or the shared price, which might not be what we want because there could be a listing with just shared rooms.

Maybe something like if (numSingles = 0 || singlePrice = 0) && (numShared = 0 || sharedPrice = 0) { return no rooms available} would be more of what you're looking for?

Reply:
That's what it was supposed to be. I forgot to commit that change after I realized that problem. Fixed!